controller/src/main/java/org/jboss/as/controller/ReadOnlyContext.java

```diff
@@ -367,4 +369,13 @@ public void deregisterCapability(String capability) {
         throw readOnlyContext();
     }
 
+    @Override
+    boolean isSkipModelValidation() {
+        return primaryContext.isSkipModelValidation();
+    }
+
+    @Override
+    void addModelValidationSteps() {
+        primaryContext.addModelValidationSteps();
```

Comment:
This is a read-only context and should not do work that is associated with writes. Probably skipModelValidation should be true.